calendar/api/parent/ext-calendar-provider.js

```diff
@@ -103,11 +103,52 @@ class ExtCalendarProvider extends cal.provider.BaseClass {
           : ["unsubscribe"];
       case "requiresNetwork":
         return !(this.capabilities.requires_network === false);
+      case "itip.transport":
+        if (this.extension.emitter.has("calendar.provider.onSend")) {
+          return this;
+        }
+        break;
     }
 
     return super.getProperty(name);
   }
 
+  scheme = "mailto";
+
+  async sendItems(aRecipients, aItipItem) {
+    let { jsmime } = ChromeUtils.import("resource:///modules/jsmime.jsm");
+    let method = aItipItem.responseMethod;
+    let transport = super.getProperty("itip.transport").wrappedJSObject;
+    let { subject, body } = transport._prepareItems(aItipItem);
+    let serializer = Cc["@mozilla.org/calendar/ics-serializer;1"].createInstance(Ci.calIIcsSerializer);
+    let itemList = aItipItem.getItemList();
+    serializer.addItems(itemList, itemList.length);
+    let methodProp = cal.getIcsService().createIcalProperty("METHOD");
+    methodProp.value = method;
+    serializer.addProperty(methodProp);
+    let icsText = serializer.serializeToString();
+    let randomGenerator = Cc["@mozilla.org/security/random-generator;1"].getService(Ci.nsIRandomGenerator);
```

Comment:
You don't need cryptographically strong randomness here. Pseudo-Random is sufficient.

Reply:
Sorry, I was just copying Thunderbird code.